go/vt/mysqlctl/backup.go

```diff
@@ -822,7 +822,7 @@ func Restore(
 	// without password, we are passing --skip-networking to greatly reduce the set
 	// of those who can connect.
 	logger.Infof("Restore: starting mysqld for mysql_upgrade")
-	err = mysqld.Start(context.Background(), "--skip-grant-tables", "--skip-networking")
+	err = mysqld.Start(context.Background(), true /* waitAsRoot */, "--skip-grant-tables", "--skip-networking")
```

Comment:
Since MySQL is started with --skip-grant-tables waiting here can be as any user, right? Maybe make this wait as dba as well, so that we have only one place waiting as root?
And since we have so many places waiting as dba and only very few waiting as root, I think it actually will be much better to not add an additional argument to Start() function, but add another function instead (e.g. StartAsRoot()). The code will be much cleaner.